src/index.js

```diff
@@ -12,17 +12,17 @@ const DEFAULT_EXTS = [ '.mjs', '.js', '.json', '.node' ];
 const readFileAsync = file => new Promise((fulfil, reject) => fs.readFile(file, (err, contents) => err ? reject(err) : fulfil(contents)));
 const statAsync = file => new Promise((fulfil, reject) => fs.stat(file, (err, contents) => err ? reject(err) : fulfil(contents)));
 const cache = fn => {
-	let cache = Object.create(null);
-	const wrapped = (s, cb = false) => {
-		if (s in cache === false) {
+	const cache = new Map();
+	const wrapped = (s, cb) => {
+		if (cache.has(s) === false) {
 			cache[s] = fn(s).catch(err => {
-				delete cache[s];
+				cache.delete(s);
 				throw err;
 			});
 		}
-		return cb ? cache[s].then(v => cb(null, v), cb) : cache[s];
+		return cache.get(s).then(v => cb(null, v), cb);
```

Comment:
Maybe we could save one microtask per cache hit if we go with the original version that also cached the "then..."?

Reply:
Could you explain further here please? I'm not sure what you're suggesting

Reply:
Ok, I confused the `cb` with `fn`. How about calling the parameters `s` and `cb` something like `param` and `done`?